test/mongoConnectionStrings.test.ts

```diff
@@ -5,55 +5,83 @@
 
 // The module 'assert' provides assertion methods from node
 import * as assert from 'assert';
-import { getDatabaseNameFromConnectionString } from '../src/mongo/mongoConnectionStrings';
+import { getDatabaseNameFromConnectionString, addDatabaseToAccountConnectionString } from '../src/mongo/mongoConnectionStrings';
 
-function testDatabaseFromConnectionString(connectionString: string, expectedDatabase: string | undefined) {
-    let database = getDatabaseNameFromConnectionString(connectionString);
-    assert.equal(database, expectedDatabase);
+function testDatabaseToAccountConnectionString(connectionString: string, databaseName: string, expectedConnectionString: string | undefined) {
+    let databaseConnectionString = addDatabaseToAccountConnectionString(connectionString, databaseName);
+    assert.equal(databaseConnectionString, expectedConnectionString);
+}
+
+function testDatabaseNameFromConectionString(connectionString: string, expectedDatabaseName: string | undefined) {
+    let databaseName = getDatabaseNameFromConnectionString(connectionString);
+    assert.equal(databaseName, expectedDatabaseName);
+    if (databaseName) {
+        testDatabaseToAccountConnectionString(connectionString, databaseName, connectionString);
```

Comment:
I'm not sure I see the value of having` testDatabaseToAccountConnectionString` inside `testDatabaseNameFromConectionString`

It seems cleaner and more reliable to have explicit/separate tests for each one

Reply:
Given the discussion, makes sense.